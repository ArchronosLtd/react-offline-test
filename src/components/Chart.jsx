import React from 'react';
import PropTypes from 'prop-types';
import { ResponsivePie } from '@nivo/pie';

// THIS DOESNT NEED TO BE A CLASS, JUST THOUGHT I'D SHOW I COULD
// DO CLASS BASED COMPONENTS TOO :)
export default class Chart extends React.Component {
    render() {
        return <div id="chart">
            <ResponsivePie data={this.props.data}
                margin={{ right: 80, left: 100 }}
                startAngle={-90}
                endAngle={90}
                enableSlicesLabels={false}
                tooltip={({ id, value }) => (
                    <span>
                        {id}: {value}%
                    </span>
                )}
            />
        </div>;
    }
}

Chart.propTypes = {
    data: PropTypes.array
}