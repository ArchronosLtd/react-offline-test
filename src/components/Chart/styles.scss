.chart {
    > .block {
        height: 50px;
        border-collapse: collapse;
        margin-bottom: 15px;
    }
}