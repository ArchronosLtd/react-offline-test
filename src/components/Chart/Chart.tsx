import React, { Component } from 'react';
import Usage from '../../models/Usage';
import Block from '../Block/Block';
import './styles.scss';

// define the props
type ChartState = {
    data: Array<any>
}

// THIS DOESNT NEED TO BE A CLASS, JUST THOUGHT I'D SHOW I COULD
// DO CLASS BASED COMPONENTS TOO :)
export default class Chart extends Component<ChartState> {
    render() {
        let blocks = this.props.data.map((usage: Usage, i: number) => {
            return <Block key={i} data={usage} />
        });

        return <div className="chart">
            {blocks}
        </div>;
    }
}