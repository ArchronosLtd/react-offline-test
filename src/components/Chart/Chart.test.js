// Link.react.test.js
import React from 'react';
import Chart from './Chart';
import renderer from 'react-test-renderer';

describe('Chart component', () => {
    const data = [
        {
            id: 'test',
            perc: 50,
            label: 'test'
        },
        {
            id: 'test2',
            perc: 50,
            label: 'test 2'
        }
    ]
    it('should have the correct id', () => {
        const component = renderer.create(
            <Chart data={data} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toBe('chart');
    });
});