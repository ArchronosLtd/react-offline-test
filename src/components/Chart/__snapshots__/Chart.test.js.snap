// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component should have the correct id 1`] = `
<div
  className="chart"
>
  <div
    className="block"
  >
    <div
      style={
        Object {
          "width": "undefined%",
        }
      }
    />
    <span>
      test
       (
      %)
    </span>
  </div>
  <div
    className="block"
  >
    <div
      style={
        Object {
          "width": "undefined%",
        }
      }
    />
    <span>
      test 2
       (
      %)
    </span>
  </div>
</div>
`;
